### ISSUE TITLE
Removal of ångström (\angstrom)

### ISSUE THREAD

**Issue by JackGin** [josephwright/siunitx#446]
Do I understand it correctly that \angstrom no longer works with a new version of siunitx? If yes, what is the alternative way to write down the Angstrom in the dimension?

**Comment by josephwright** [comment]
The BIPM have removed ångström (see file:///Users/joseph/Downloads/SI-Brochure-9-EN.pdf for the accepted units), which means I have from `siunitx`. It's easy enough to add manually
```latex
\DeclareSIUnit\angstrom{\text{Å}}
```
or if you are using `fontspec` more simply
```latex
\DeclareSIUnit\angstrom{Å}
```

**Comment by JackGin** [comment]
It is a pity because it breaks the backward compatibility with thousands, if not millions, of papers in physics written so far.

**Comment by josephwright** [comment]
> It is a pity because it breaks the backward compatibility with thousands, if not millions, of papers in physics written so far.

'It's not my fault': the BIPM _define_ the SI, I just try to stick as close as possible to it. I'm not particularly happy that they've removed ångströms as I use them routinely myself, but that's what's happened. (As a matter of policy, I try hard to avoid making any personal judgements on units: `siunitx` is supposed to implement the rules set by BIPM, informed by other bodies such as NIST, not to set any rules itself.)

On 'existing documents' - also all of them will either be published (so have an HTML/PDF version of record) or using a TeX system which is reflective of their age.

**Comment by JackGin** [comment]
Do you consider including some additional units in the package, clearly designated as such in the package documentation? BIPM is BIPM, but siuintx may have its own goals, the user convenience included? Just my two cents.

**Comment by rafaelpap** [comment]
It's wrong to use?

\DeclareSIUnit\angstrom{\SIUnitSymbolAngstrom}

**Comment by josephwright** [comment]
> It's wrong to use?
> 
> \DeclareSIUnit\angstrom{\SIUnitSymbolAngstrom}

Nothing wrong, it's just that I decided in the v3 work that `\SIUnitSymbol...` were an indirection that didn't really help - they are there for compatibility

**Comment by josephwright** [comment]
> Do you consider including some additional units in the package, clearly designated as such in the package documentation? BIPM is BIPM, but siuintx may have its own goals, the user convenience included? Just my two cents.

In the distance past (v1.x) I did provide a range of 'other' units, but the issue was that this was an open-ended thing: decided what 'counts' was very hard. For v2 I took a much harder line

- Units explicitly in the SI Brochure
-  Percent (more-or-less covered by the SI Brochure)
- 'Binary' prefixes and associated units - pointed to by the SI Brochure but not formally part of SI

That way things are nice and clear - I'm not making a judgement on what counts as 'additions'.

I can see that removals by BIPM are a bit different, as they _used_ to support ångström (also bar, plus things like hartree). However, we still have that question of what counts - depending on where one looks, there are other units that used to be included.

One can always get the older behaviour strictly with
```latex
\usepackage{siunitx}[=v2]
```
or go with a simple
```latex
\DeclareSIUnit\angstrom{\text{Å}}
```
which works with v2 and v3 (much as I have to for say 'molar' anyway).

I will think a bit more about this, but I'm very wary of adding units beyond a carefully defined set.

**Comment by JackGin** [comment]
>\usepackage{siunitx}[=v2]
That's nice!
> I'm very wary of adding units beyond a carefully defined set
As a physicist to physicist — why?:) We are talking here about the package to write our own papers (where we freely use Gauss units and what not), not the national standards.

**Comment by rafaelpap** [comment]
Well, allow me to disagree. I expect a package named siunitx to deal with SI units.

**Comment by josephwright** [comment]
@JackGin  I'm a chemist ;) More seriously, whilst `siunitx` is designed to support whatever units people choose (I almost always need `\DeclareSIUnit\Molar{\textsc{m}}`), 'out of the box' it's only meant to provide an 'official' set of units.

**Comment by josephwright** [comment]
In particular, if you read the appendix to toe SI Brochure (https://www.bipm.org/en/publications/si-brochure), you'll see there are a _lot_ of things that have been removed over time.

**Comment by aoconnor-ufl** [comment]
I believe you 1) recommend that users explicitly set the version to ensure future updates don't break compatibility with currently accepted units and 2) you should include a user-visible warning if that version number is not set.  See the mhchem package for a good example on how to implement this.

I just happened to be using the barn unit in my papers, and now must update the documents even though nothing has changed except the package version.

**Comment by josephwright** [comment]
@aoconnor-ufl The issue I have with that approach is twofold. First, versions of packages should be about their API, not about wider standards. The second is that specifying a version without reason in sources will tie people to older code even where it should work fine.

That said, I can think of a couple of adjustments. I'm wondering whether I should have a setting for the BIPM Brochure version to load pre-defined units. (I'm not 100%: the whole point of having a simple mechanism to create them is they can be in the preamble of a document.) The second is that I think I can see a way to have the units in a deprecated form with appropriate warning - that should satisfy existing users while also making it clear that, like it or not, BIPM have removed some units from the accepted set.

**Comment by Canageek** [comment]
Might I recommend a compromise: Don't remove units that have been in the SI brochure once they have been in siunitx, but don't remove ones that have already been added. 

That way you don't break things on poor grad students in the middle of writing their thesis. 

Also the one compile warning doesn't help at all if you are using latexmk and it automatically compiles your document multiple times. The first warning I got was it breaking everything.

**Comment by josephwright** [comment]
@Canageek As I've said elsewhere, the 'new' SI is not a recent thing - it's been heavily trailed for several years and has been the current set of units for about three. Yes, I know that most people don't bother to check what the current unit position actually is, but since I released of v2 of `siunitx` in 2009 I've been clear that the set of units is not 'up to me' but rather 'those that BIPM define as an international standard'. Moreover, adding non-standard units is meant to be easy.

On warnings, there is not much I can do - I'd hope that automated systems make these available to the user. (Warnings, unlike info messages, are things the user is definitely supposed to address.) 

You'll see that I've adjusted the position to make the unit macros continue to work but with a warning that they should be locally-defined. What I am very mindful of is that for new _documents_, people should be adding `\DeclareSIUnit \bar ...` or similar: whatever the rights or wrongs of the BIPM decision, the units I removed are no longer part of the standard. If I don't warn people, then they'll not do that. (That also means, of course, that we should all be thinking about making their definitions explicit in our work.)

**Comment by josephwright** [comment]
To be clear, I use ångström and bar in my day-to-day research work, so it's not like I'm _happy_ with BIPM for removing these (although I do see the logic, as they are easy enough to convert to picometres and pascals, respectively). But all that means is that they are now like 'molar' (`\textsc{m}`) - a non-standard unit I'll need to add to my preamble.

**Comment by u-fischer** [comment]
@Canageek 

> That way you don't break things on poor grad students in the middle of writing their thesis.

"poor grad students" should be happy that someone warns them that they use deprecated units. That is not something one should have in a thesis unless the supervisor explicitly asks for it as it shows a lack of research.

**Comment by JackGin** [comment]
> > That way you don't break things on poor grad students in the middle of writing their thesis.
> 
> "poor grad students" should be happy that someone warns them that they use deprecated units. That is not something one should have in a thesis unless the supervisor explicitly asks for it as it shows a lack of research.

With all due respect, this is utter BS. Please do not save the souls of the grad students unless they ask you to. Besides, how is that using non-SI units might show a lack of research??? The wind blows wherever it pleases.

**Comment by josephwright** [comment]
@u-fischer  I wouldn't go that far - we are talking here about units which are in broad 'circulation' and which were accepted for use in the 8th Edition of the 'rules'.

The balance is tricky - if you look back over previous BIPM documents, there are a lot of units that were removed and have become much less common, and some that have more stuck around. I suspect that ångström and bar will both fall into the latter category, while the 'natural' units I am less sure about (as 'new' SI after all has some of them as fixed constants).

**Comment by JackGin** [comment]
@rafaelpap 
This was a strong disagreement with the statement, but not insulting by any means. As to the standards, the whole point is that in original scientific research following 'standards' is not mandatory, and moreover may very well be counterproductive. Besides, the de-facto standards may vary depending on the field of the research. Say, in quantum electrodynamics you normally use CGS, not SI at all.

**Comment by Canageek** [comment]
> 
> 
> @Canageek
> 
> > That way you don't break things on poor grad students in the middle of writing their thesis.
> 
> "poor grad students" should be happy that someone warns them that they use deprecated units. That is not something one should have in a thesis unless the supervisor explicitly asks for it as it shows a lack of research.

Angstrom are the standard for all of crystallography, so I'd tell anyone who told me to stop using them they are very welcome to rewrite every piece of crystallogrpahy software written in the last 50 years, and then justify to every future researcher why they have to do mental conversions for every paper published in Crystal Growth and Design, IUCrJ, all the Acata Cryst. jouranls, etc from now on. Plus it is the general unit used in chemistry as it is a lot more convenient then pm. So yeah, I'd have some very rude words for anyone who implied I didn't know what I am talking about by using them, and imply that it shows they don't actually know what they are talking about.

**Comment by Canageek** [comment]
> 
> 
> @Canageek As I've said elsewhere, the 'new' SI is not a recent thing - it's been heavily trailed for several years and has been the current set of units for about three. Yes, I know that most people don't bother to check what the current unit position actually is, but since I released of v2 of `siunitx` in 2009 I've been clear that the set of units is not 'up to me' but rather 'those that BIPM define as an international standard'. Moreover, adding non-standard units is meant to be easy.
> 
> On warnings, there is not much I can do - I'd hope that automated systems make these available to the user. (Warnings, unlike info messages, are things the user is definitely supposed to address.)
> 
> You'll see that I've adjusted the position to make the unit macros continue to work but with a warning that they should be locally-defined. What I am very mindful of is that for new _documents_, people should be adding `\DeclareSIUnit \bar ...` or similar: whatever the rights or wrongs of the BIPM decision, the units I removed are no longer part of the standard. If I don't warn people, then they'll not do that. (That also means, of course, that we should all be thinking about making their definitions explicit in our work.)

They do make the warnings available, but they automatically start a second compile if page numbers or references have changed. So, first compile displays the warning, latexmk detects that page numbers have change and starts a compile, and then it fails since you didn't fix the warning, despite the chance you didn't have time to do so.

**Comment by josephwright** [comment]
> > @Canageek
> > > That way you don't break things on poor grad students in the middle of writing their thesis.
> > 
> > 
> > "poor grad students" should be happy that someone warns them that they use deprecated units. That is not something one should have in a thesis unless the supervisor explicitly asks for it as it shows a lack of research.
> 
> Angstrom are the standard for all of crystallography, so I'd tell anyone who told me to stop using them they are very welcome to rewrite every piece of crystallogrpahy software written in the last 50 years, and then justify to every future researcher why they have to do mental conversions for every paper published in Crystal Growth and Design, IUCrJ, all the Acata Cryst. jouranls, etc from now on. 

That of course is all true but an argument that I assume was had in drawing up the latest SI Brochure. No one is saying people won't continue to use 'non-approved' units, it's about which ones I can have pre-defined and which need to be in people's sources.

(People do change the units they use over time: I'm not saying it will happen here but we also can't rule out that it might. At least we are only talking about a pretty clear factor of 10 issue here: much more troublesome are non-integer or 'hard to spot' changes. As a chemist, I guess I'd point to the change from tau to delta for NMR shift as 'borderline mind-bending', but that's looking back at older literature and finding it hard to follow.)

**Comment by josephwright** [comment]
> > @Canageek As I've said elsewhere, the 'new' SI is not a recent thing - it's been heavily trailed for several years and has been the current set of units for about three. Yes, I know that most people don't bother to check what the current unit position actually is, but since I released of v2 of `siunitx` in 2009 I've been clear that the set of units is not 'up to me' but rather 'those that BIPM define as an international standard'. Moreover, adding non-standard units is meant to be easy.
> > On warnings, there is not much I can do - I'd hope that automated systems make these available to the user. (Warnings, unlike info messages, are things the user is definitely supposed to address.)
> > You'll see that I've adjusted the position to make the unit macros continue to work but with a warning that they should be locally-defined. What I am very mindful of is that for new _documents_, people should be adding `\DeclareSIUnit \bar ...` or similar: whatever the rights or wrongs of the BIPM decision, the units I removed are no longer part of the standard. If I don't warn people, then they'll not do that. (That also means, of course, that we should all be thinking about making their definitions explicit in our work.)
> 
> They do make the warnings available, but they automatically start a second compile if page numbers or references have changed. So, first compile displays the warning, latexmk detects that page numbers have change and starts a compile, and then it fails since you didn't fix the warning, despite the chance you didn't have time to do so.

I can't do much there other than point out that warnings are in every run for a deprecated unit, which means the final `jobname.log` will have exactly the same warning as the first one. I hope people do read their logs - package authors add warnings for a reason. (I know that the reasons vary and some can be quite annoying-and-pointless.)

**Comment by Canageek** [comment]
> 
> 
> > > @Canageek
> > > > That way you don't break things on poor grad students in the middle of writing their thesis.
> > > 
> > > 
> > > "poor grad students" should be happy that someone warns them that they use deprecated units. That is not something one should have in a thesis unless the supervisor explicitly asks for it as it shows a lack of research.
> > 
> > 
> > Angstrom are the standard for all of crystallography, so I'd tell anyone who told me to stop using them they are very welcome to rewrite every piece of crystallogrpahy software written in the last 50 years, and then justify to every future researcher why they have to do mental conversions for every paper published in Crystal Growth and Design, IUCrJ, all the Acata Cryst. jouranls, etc from now on.
> 
> That of course is all true but an argument that I assume was had in drawing up the latest SI Brochure. No one is saying people won't continue to use 'non-approved' units, it's about which ones I can have pre-defined and which need to be in people's sources.
> 
> (People do change the units they use over time: I'm not saying it will happen here but we also can't rule out that it might. At least we are only talking about a pretty clear factor of 10 issue here: much more troublesome are non-integer or 'hard to spot' changes. As a chemist, I guess I'd point to the change from tau to delta for NMR shift as 'borderline mind-bending', but that's looking back at older literature and finding it hard to follow.)

Given how their 'redefine things in terms of a physical constant" thrust ignores all future improved measurements to said physical constant, I do not assume anything they do makes sense or is a good idea at this point.

Also: Sounds like the paper I read that reported the same measurement on various compound microns, nanometers and wavenumbers.

**Comment by Canageek** [comment]
> 
> 
> > > @Canageek As I've said elsewhere, the 'new' SI is not a recent thing - it's been heavily trailed for several years and has been the current set of units for about three. Yes, I know that most people don't bother to check what the current unit position actually is, but since I released of v2 of `siunitx` in 2009 I've been clear that the set of units is not 'up to me' but rather 'those that BIPM define as an international standard'. Moreover, adding non-standard units is meant to be easy.
> > > On warnings, there is not much I can do - I'd hope that automated systems make these available to the user. (Warnings, unlike info messages, are things the user is definitely supposed to address.)
> > > You'll see that I've adjusted the position to make the unit macros continue to work but with a warning that they should be locally-defined. What I am very mindful of is that for new _documents_, people should be adding `\DeclareSIUnit \bar ...` or similar: whatever the rights or wrongs of the BIPM decision, the units I removed are no longer part of the standard. If I don't warn people, then they'll not do that. (That also means, of course, that we should all be thinking about making their definitions explicit in our work.)
> > 
> > 
> > They do make the warnings available, but they automatically start a second compile if page numbers or references have changed. So, first compile displays the warning, latexmk detects that page numbers have change and starts a compile, and then it fails since you didn't fix the warning, despite the chance you didn't have time to do so.
> 
> I can't do much there other than point out that warnings are in every run for a deprecated unit, which means the final `jobname.log` will have exactly the same warning as the first one. I hope people do read their logs - package authors add warnings for a reason. (I know that the reasons vary and some can be quite annoying-and-pointless.)

I would suggest doing this, *with* the code to copy and paste into your document to make the error go away.

**Comment by josephwright** [comment]
> > I can't do much there other than point out that warnings are in every run for a deprecated unit, which means the final `jobname.log` will have exactly the same warning as the first one. I hope people do read their logs - package authors add warnings for a reason. (I know that the reasons vary and some can be quite annoying-and-pointless.)
> 
> I would suggest doing this, _with_ the code to copy and paste into your document to make the error go away.

I'm confused: the warning _is_ there on every run. As to what you should add to a document - that rather depends. (My 'out of the box' definitions for natural units, for example, make quite a few assumptions that may or may not hold.)

**Comment by josephwright** [comment]
> Given how their 'redefine things in terms of a physical constant" thrust ignores all future improved measurements to said physical constant, I do not assume anything they do makes sense or is a good idea at this point.

Definitely a discussion for another place: I'm no metrology expert.

**Comment by josephwright** [comment]
> > > > I can't do much there other than point out that warnings are in every run for a deprecated unit, which means the final `jobname.log` will have exactly the same warning as the first one. I hope people do read their logs - package authors add warnings for a reason. (I know that the reasons vary and some can be quite annoying-and-pointless.)
> > > 
> > > 
> > > I would suggest doing this, _with_ the code to copy and paste into your document to make the error go away.
> > 
> > 
> > I'm confused: the warning _is_ there on every run. As to what you should add to a document - that rather depends. (My 'out of the box' definitions for natural units, for example, make quite a few assumptions that may or may not hold.)
> 
> Sorry, I thought yous said it only warned on the first run, then through and error.
> 
> Couldn't you print out the code their document is _currently_ using?

There's not a lot of space in a normal warning (one or two lines). Moreover, I hope people do know their own units!

**Comment by Canageek** [comment]
I know my units, but I have no idea how to tell siunitx to print the angstrom symbol! 

Also, really? When I used the lastpage it bugged me with something like a 5 line error message until I moved to pageslts (despite it being more of a pain to set up). 

Also aren't most unit definitions one line? 
Something like: "WARNING: Angstrome are no longer an SI unit. To keep using them add "\DeclareSIUnit\angstrom{Å}" to your preamble. But you could autodetect the current definition if you are worried about giving them the wrong command to use?

**Comment by josephwright** [comment]
@Canageek  Well ideally I'd keep warnings short and have out-and-out errors longer, but that's probably not what you are after :) I've adjusted the warning so you get the full context: what definition I'm using, so you can copy-paste it if you like.

**Comment by Canageek** [comment]
Oh, that sounds perfect to me actually. You were just worried about the definitions you use having issues.

**Comment by josephwright** [comment]
I think the adjustments made address the issue: users get a warning and can then decide what to do.
